makefiles/version.mk

```diff
@@ -1,6 +1,6 @@
 ##### version
 NCCL_MAJOR   := 2
-NCCL_MINOR   := 4
-NCCL_PATCH   := 8
+NCCL_MINOR   := 5
+NCCL_PATCH   := 0
```

Comment:
We don't expect NCCL to have a 2.5? That may end up confusing if they release 2.5 w/o bf16 support?

Reply:
If bf16 support is going to be added as an extra ROCm feature to an otherwise compatible NCCL 2.4.8 API, I second this concern.  Can we instead add a preprocessor symbol to the rccl.h header indicating bf16 support?  Then applications could check for this symbol to determine if the feature is available, rather than checking an artificial version number that now diverges from upstream.

Reply:
Removed changes to API version.
Added #define RCCL_BFLOAT16 1 in rccl.h instead